### PR TITLE
Allow non saved fields for existing samples to be optional

### PR DESCRIPTION
**Pull request by patrikgrenfeldt**
This PR fixes the problem to submit orders with families that you have added existing samples to.

How to test:
0. install on stage (add instructions on how to as needed)
1. run test in https://github.com/Clinical-Genomics/cgweb/pull/98

Expected outcome:
Success

### CODE REVIEW

File: cg/meta/orders/schema.py [hunk 1]
```diff
@@ -110,22 +110,22 @@ def validate(self, value):
 
     # This information is required
     'name': validators.RegexValidator(NAME_PATTERN),
-    'container': validators.Any(CONTAINER_OPTIONS),
+    'container': validators.Optional(TypeValidator(str, allow_none=True), None),
```
Comment (ingkebil) [inline cg/meta/orders/schema.py]: Should prolly move to the "Order portal specific" block on line 109

File: cg/meta/orders/schema.py [hunk 2]
```diff
@@ -110,22 +110,22 @@ def validate(self, value):
...
     'data_analysis': str,
     'application': str,
     'sex': validators.Any(SEX_OPTIONS),
     'family_name': validators.RegexValidator(NAME_PATTERN),
     'require_qcok': bool,
     'tumour': bool,
-    'source': str,
+    'source': validators.Optional(TypeValidator(str, allow_none=True), None),
```
Comment (ingkebil) [inline cg/meta/orders/schema.py]: Should prolly move to the "Order portal specific" block on line 109

File: cg/meta/orders/schema.py [hunk 3]
```diff
@@ -110,22 +110,22 @@ def validate(self, value):
...
     'priority': validators.Any(PRIORITY_OPTIONS),
 
     # Required if Plate
     'container_name': validators.Optional(str, None),
     'well_position': validators.Optional(TypeValidator(str, allow_none=True), None),
 
     # This information is required for Balsamic analysis (cancer)
-    'capture_kit': validators.Any(CAPTUREKIT_CANCER_OPTIONS),
+    'capture_kit': validators.Optional(TypeValidator(str, allow_none=True), None),
```
Comment (ingkebil) [inline cg/meta/orders/schema.py]: hehe, it's not required when it's optional ;) Don't have a good suggestion on how to solve this tho

File: cg/meta/orders/schema.py [hunk 4]
```diff
@@ -152,21 +152,21 @@ def validate(self, value):
 
     # required
     'name': validators.RegexValidator(NAME_PATTERN),
-    'capture_kit': validators.Any(CAPTUREKIT_OPTIONS),
+    'capture_kit': validators.Optional(TypeValidator(str, allow_none=True), None),
```
Comment (ingkebil) [inline cg/meta/orders/schema.py]: optional, not required, should prolly move to an optional block

File: cg/meta/orders/schema.py [hunk 5]
```diff
@@ -152,21 +152,21 @@ def validate(self, value):
...
 
     # Required if data analysis in Scout
     'panels': ListValidator(str, min_items=0),
     # todo: find out if "Additional Gene List" is "lost in translation", implement in OP or remove from OF
     'status': validators.Any(STATUS_OPTIONS),
 
     # Required if samples are part of trio/family
-    'mother': validators.Optional(validators.RegexValidator(NAME_PATTERN), None),
-    'father': validators.Optional(validators.RegexValidator(NAME_PATTERN), None),
+    'mother': validators.Optional(TypeValidator(str, allow_none=True), None),
```
Comment (ingkebil) [inline cg/meta/orders/schema.py]: So, this used to be a regexvalidator with a pattern. Now it is an optional string. Does that mean we cannot have an optional regex?

File: cg/meta/orders/schema.py (hunk 1 above)
Comment (patrikgrenfeldt) [reply cg/meta/orders/schema.py]: The meaning of "Order portal specific" in this context is whether the attribute exists in the Orderform or not and thus I think it is placed under the correct comment? 

What might be a bit confusing here is that some fields are required when the information comes from the Orderform but is optional when you select an already existing sample in the Orderportal.  Maybe we should implement something along the lines of OrValidator( 'internal_id', 'container' ) ?

File: cg/meta/orders/schema.py (hunk 2 above)
Comment (patrikgrenfeldt) [reply cg/meta/orders/schema.py]: I think I will update the comments..

File: cg/meta/orders/schema.py (hunk 5 above)
Comment (patrikgrenfeldt) [reply cg/meta/orders/schema.py]: Implemented now :)

**Comment by patrikgrenfeldt** [comment]
@ingkebil can you review it again, the changes and comments that I made?

**Comment by emiliaol** [comment]
Also tested with the cgweb PR with great success!

**Pull request merged**